features/270-next-skycli/api.md

```diff
@@ -0,0 +1,168 @@
+## Feature Overview
+
+In next, we will have cli tools for user to manage the cluster's applications.
+Since we will only have skycli in the first version, skycli should be able to
+do almost all the operations of v1 portal.
+
+The following spec will cover the EE version first.
+
+
+## User scenarios
+
+- Cluster admin can create cluster user
+- Cluster admin assign user to app
+- Cluster user can create app
+- Cluster user can view, update and delete their apps config
+- Cluster user deploy CF to their apps
+- Cluster user can update apps custom domain setting, add new custom domain to app
+- Cluster admin can do everything in the cluster
+
+## Commands naming guidelines
+
+**Format**
+
+```
+skycli [group] [verb/action] [parameters]
+```
+
+**Some common operations**
+
+| Operation | Verb        |
+|-----------|-------------|
+| Create    | create      |
+| Read      | view / list |
+| Update    | update      |
+| Delete    | remove      |
+
+
+## Commands
+
+- skycli auth
+    - [skycli auth login](commands.md#skycli-auth-login)
+    - [skycli auth logout](commands.md#skycli-auth-logout)
+
+- skycli config
+    - [skycli config view](commands.md#skycli-config-view)
+    - [skycli config set-app [APP_NAME]](commands.md#skycli-config-set-app)
+    - [skycli config set-cluster-server [https://1.2.3.4]](commands.md#skycli-config-set-cluster-server)
+
+- skycli user (Admin only, cluster user management)
+    - [skycli user create](commands.md#skycli-user-create)
+    - [skycli user list](commands.md#skycli-user-list)
+
+#### Application commands
+
+- skycli app
+    - [skycli app list](commands.md#skycli-app-list)
+    - [skycli app create](commands.md#skycli-app-create)
+    - [skycli app add-user [USER_EMAIL]](commands.md#skycli-app-add-user)
+    - [skycli app view-tenant-config](commands.md#skycli-app-view-config)
+    - [skycli app update-tenant-config -f [TENANT_CONFIG_YAML_FILE]](commands.md#skycli-app-update-config)
+
+- skycli domain
+    - [skycli domain add [CUSTOM_DOMAIN]](commands.md#skycli-domain-create)
+    - [skycli domain verify [CUSTOM_DOMAIN]](commands.md#skycli-domain-verify)
+    - [skycli domain update [CUSTOM_DOMAIN]](commands.md#skycli-domain-update)
+    - [skycli domain list](commands.md#skycli-domain-list)
+
+- skycli cf
+    - [skycli cf list](commands.md#skycli-cf-list)
+    - [skycli cf deploy [FUNCTION_NAME]](commands.md#skycli-cf-deploy)
+    - [skycli cf logs [FUNCTION_NAME]](commands.md#skycli-cf-logs)
+    - [skycli cf invoke [FUNCTION_NAME] --payload [PAYLOAD_JSON]](commands.md#skycli-cf-invoke)
+
+- skycli secret
+    - [skycli secret list](commands.md#skycli-secret-list)
+    - [skycli secret create [SECRET_NAME] [SECRET_VALUE]](commands.md#skycli-secret-create)
+    - [skycli secret update [SECRET_NAME] [SECRET_VALUE]](commands.md#skycli-secret-update)
```

Comment:
To support key rotation, the stupid way is setup 2 secrets: `SECRET1` and `SECRET2` and up to the admin to replace either one, however I see most platform do not show the last update time, so it is very easy to replace the wrong one. We may do better on this or even handle the rotation is more human way.

Reply:
Key rotation is a common use case that we should consider! We are discussing app versioning in another thread, which provide a more human way to handle rotation. (https://github.com/SkygearIO/features/pull/278)
For skycli, I think we should show the last update time to avoid human error too.
Thanks! @hidden-dimensions